pkg/servicescm/servicescm.go

```diff
@@ -47,11 +47,12 @@ type NodeCmdArg struct {
 	NodeObjectJsonPath string `json:"nodeObjectJsonPath"`
 }
 
-// PowershellCmdArg describes a PowerShell variable and how its value can be populated
-type PowershellCmdArg struct {
-	// Name is the variable name as it appears in commands
-	Name string `json:"name"`
-	// Path is the location of a PowerShell script whose output is the value of the variable
+// PowershellPreScript describes a PowerShell script to be ran and an optional variable to be populated
+type PowershellPreScript struct {
```

Comment:
If you are thinking about `PowerShellPostScript` shouldn't this be generic?

Reply:
PowerShellPostScript would just need to be a []string, as there wouldn't be any variables to replace, given it runs after the service.

Reply:
Would this also be the case for arbitrary scripts that we were going to introduce instead of always having scripts associated with a service? While there might not be variables to replace there either do you think the struct could be reused with other members added to it?